companion/src/storage/sdcard.cpp

```diff
@@ -82,6 +82,28 @@ bool SdcardFormat::deleteFile(const QString & filename)
   return true;
 }
 
+// Check if the connected radio has more models than the currently loaded set.
+bool SdcardFormat::radioHasExtraModels(const RadioData & radioData)
+{
+  RadioData tmpRadioData;
+  load(tmpRadioData);
+  return tmpRadioData.models.size() > radioData.models.size();
+}
```

Comment:
Noticeable performance hit when lots of models, as all the model files on the radio have to be read, parsed, decoded and loaded into memory variable (including those just written) just to get a count. Why not just count the model files?